### DISCUSSION TITLE
Is it possible to get a H264 stream from screen data?

### DISCUSSION THREAD

**Discussion by Saddingt0n** [FreeRDP/FreeRDP#8500] [Q&A] [answered]
I will be talking about X11 client. My limited understanding is that during a RDP session. Whenever the screen of the remote machine changes, the X11 client will update the bitmap parts of the changed section only.

I want to forward the screen data from X11 client or sample client to another app of mine. I get the screen data from `gdi->primary_buffer` in `tf_end_paint` function and sends that through socket. But this seems to perform poorly. I notice the frame rate to be low.

I'm not experienced in video streaming and I have worked with a H264 stream before so I'm wondering if I can get a H264 stream of screen data from FreeRDP?

**Comment by akallabeth** [comment, marked as answer]
well, you have a bitmap in `gdi->primary_buffer` and a trigger `tf_end_paint` so all you need to do is feed the bitmap to your h264 encoder of choice.
you might even find some example with the freerdp encoder helper functions
